### PR TITLE
Migrate observability, logging, cloudwatch and secured cluster secrets to External Secrets Operator

### PR DESCRIPTION
**Pull request by kovayur**
## Description
Migrate observability, logging, cloudwatch and secured cluster secrets to External Secrets Operator

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

## Test manual
Can be fully tested only on integration

### CODE REVIEW

File: dp-terraform/helm/rhacs-terraform/charts/cloudwatch/templates/01-operator-02-secret-dead-mans-switch.yaml [hunk 1]
```diff
@@ -1,9 +1,21 @@
-apiVersion: v1
-kind: Secret
+apiVersion: external-secrets.io/v1beta1
```
Comment (stehessel) [inline dp-terraform/helm/rhacs-terraform/charts/cloudwatch/templates/01-operator-02-secret-dead-mans-switch.yaml:1]: I think the name for this file is really off 🤦. Will have to fix that separately...

File: dp-terraform/helm/rhacs-terraform/charts/logging/templates/01-logging-02-secret-cloudwatch.yaml [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+    name: {{ .Values.global.secretStore.aws.secretsManagerSecretStoreName }}
+    kind: ClusterSecretStore
+  target:
+    name: cloudwatch
+    creationPolicy: Owner
+    template:
+      type: Opaque
+      engineVersion: v2
+      data:
+        aws_access_key_id: {{ printf "'{{ .aws_access_key_id | b64enc }}'" }}
```
Comment (stehessel) [inline dp-terraform/helm/rhacs-terraform/charts/logging/templates/01-logging-02-secret-cloudwatch.yaml:17]: It's already there, but do you know why there is double base64 encoding?

**Review by stehessel — APPROVED** [review]

File: dp-terraform/helm/rhacs-terraform/charts/logging/templates/01-logging-02-secret-cloudwatch.yaml (hunk 2 above)
Comment (kovayur) [reply dp-terraform/helm/rhacs-terraform/charts/logging/templates/01-logging-02-secret-cloudwatch.yaml]: `data` must be encoded. and when using `stringData` the encoding is applied automatically.
It was a mistake, External Secrets Operator behaves as `stringData` and applies the encoding automatically.

File: dp-terraform/helm/rhacs-terraform/templates/fleetshard-sync-secret.yaml [hunk 3]
```diff
@@ -41,7 +41,7 @@ metadata:
   namespace: {{ .Release.Namespace }}
 spec:
   secretStoreRef:
-    name: {{ .Values.secretStore.aws.parameterStoreSecretStoreName }}
+    name: {{ .Values.global.secretStore.aws.parameterStoreSecretStoreName }}
```
[the remark this answers is not in the document]
Comment (kovayur) [reply dp-terraform/helm/rhacs-terraform/templates/fleetshard-sync-secret.yaml:44]: Some non-sensitive parameters created by terraform are still stored in Parameter Store.

**Comment by kovayur** [comment]
@stehessel I tested the changed secrets on integration and applied some fixes, including the curly braces escape in alertmanager.yaml

**Pull request merged**